packages/pwa-kit-dev/src/utils/logger.js

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * Array defining the order of log levels from lowest to highest severity.
+ * The order of log levels is important. It determines the logging threshold.
+ * @type {string[]}
+ */
+const LOG_LEVELS = ['debug', 'info', 'warn', 'error']
+
+// Default log level for the logging threshold. It can be any of the LOG_LEVELS values.
+const DEFAULT_LOG_LEVEL = 'info'
+
+// Default log format to use if not explicitly provided. It can be 'JSON' or 'TEXT'.
+const DEFAULT_LOG_FORMAT = 'JSON'
+
+const isServerSide = typeof window === 'undefined'
+
+// Configuration object defining logging settings for server/client environments
+const loggerConfig = {
```

Comment:
This object kinda looks similar to the arguments for the constructor of your logger class. Have you thought about this object actually being the default config? and add in the names space to it? Because it seems like that is what it's being used for.